editoast/editoast_derive/src/error.rs

```diff
@@ -37,15 +37,27 @@ pub fn expand_editoast_error(input: &DeriveInput) -> Result<TokenStream> {
         _ => return Err(Error::custom("EditoastError: Only enums are supported.")),
     };
     let variants = parse_variants(enum_data)?;
+    let default_status = options.default_status.unwrap_or(DEFAULT_STATUS_CODE);
+    let get_statuses = expand_get_statuses(&variants, default_status)?;
 
-    let get_statuses = expand_get_statuses(
-        &variants,
-        options.default_status.unwrap_or(DEFAULT_STATUS_CODE),
-    )?;
     let contexts = expand_contexts(&variants);
-    let get_types = expand_get_types(&variants, base_id.clone());
+    let get_types = expand_get_types(&variants, &base_id);
+
+    let (error_definition, errors): (Vec<Result<TokenStream>>, Vec<Result<TokenStream>>) = variants
+        .iter()
+        .map(|v| parse_error_definition(&base_id, default_status, v))
+        .partition(Result::is_ok);
+
+    if !errors.is_empty() {
+        return errors.first().unwrap().to_owned();
+    }
```

Comment:
See this [documentation](https://doc.rust-lang.org/rust-by-example/error/iter_result.html#fail-the-entire-operation-with-collect)

```suggestion
    let error_definition = variants
        .iter()
        .map(|v| parse_error_definition(&base_id, default_status, v))
        .collect::<Result<Vec<TokenStream>>>()?;

    let error_definitions: TokenStream = error_definition.into_iter().collect();
```